nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/TicketQuery.java

```diff
@@ -0,0 +1,294 @@
+        String query = createPlainQueryString();
+        Map<Integer, Object> parameters = createParameterReplacementList();
+
+        PreparedStatement statement = connection.prepareStatement(query);
+        parameters.entrySet().forEach(entry -> {
+            int parameterIndex = entry.getKey();
+            Object value = entry.getValue();
+            try {
+                //This currently only contains the values used by the columns in the Tickets table, add more if required.
+                if (value == null) {
+                    statement.setNull(parameterIndex, Types.NULL);
+                } else if (value instanceof Integer) {
+                    statement.setInt(parameterIndex, (int) value);
+                } else if (value instanceof UUID) {
+                    statement.setString(parameterIndex, value.toString()); //Since in our DB Tables we are representing a UUID as a 36 character string, set a string value.
+                } else if (value instanceof Instant) {
+                    statement.setTimestamp(parameterIndex, Timestamp.from((Instant) value));
+                } else if (value instanceof Boolean) {
+                    statement.setBoolean(parameterIndex, (boolean) value);
+                }
+            } catch (SQLException ex) { }
+        });
+
+        return statement;
+    }
+
+    /**
+     * Creates an SQL query string with replacement parameters using the query filters provided by the {@link Builder}.
+     *
+     * @return An SQL query.
+     */
+    public String createPlainQueryString() {
+        String query = "SELECT * FROM Tickets";
+        if (queries.size() > 0) {
+            query += " WHERE ";
+            Iterator<Map.Entry<String, Map<Integer, Object>>> entryIterator = queries.entrySet().iterator();
+            while (entryIterator.hasNext()) {
+                Map.Entry<String, Map<Integer, Object>> argument = entryIterator.next();
+
+                //Add the query filter to the SQL query.
+                query += "(" + argument.getKey() + ")";
+                if (entryIterator.hasNext()) query += " AND ";
+            }
+        }
+        query += ");";
+
+        return query;
+    }
+
+    /**
+     * Creates a map containing parameters mapped to the value which needs to be replaced in the final query string - in
+     * order.
+     *
+     * @return A map of replacement parameters.
+     */
+    public Map<Integer, Object> createParameterReplacementList() {
+        Map<Integer, Object> replacementParameters = Maps.newHashMap();
+
+        for (Map<Integer, Object> singleQueryParameters : queries.values()) {
+            int baseSize = replacementParameters.size();
+            singleQueryParameters.entrySet().forEach(qp -> replacementParameters.put(baseSize + qp.getKey(), qp.getValue()));
+        }
+
+        return ImmutableMap.copyOf(replacementParameters);
+    }
+
+    public static Builder builder() {
+        return new Builder();
```

Comment:
I think Sponge has a store that we can put builder constructors into so we can make the builder an interface, otherwise we'll create our own.